app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -155,27 +187,26 @@ class HikesViewModel(
    * Downloads the current user's saved hikes from the database and caches them locally.
    *
    * This function updates the saved status of the loaded hikes in [hikeFlows]. If saved hikes are
-   * currently loaded (a call to [loadSavedHikes] has been made and no other loading call has been
-   * performed since), calling this function will update the whole loaded hikes list to match it
-   * with the new saved hikes.
+   * currently loaded (see [loadedHikesType]), calling this function will update the whole loaded
+   * hikes list to match it with the new saved hikes.
    *
    * @param onSuccess To be called when the saved hikes cache has been updated successfully.
    * @param onFailure Will be called if an error is encountered.
    */
   fun refreshSavedHikesCache(onSuccess: () -> Unit = {}, onFailure: () -> Unit = {}) =
       viewModelScope.launch {
         // Let the user know a heavy load operation is being performed
-        _loading.value = true
+        setLoading(true)
 
-        val success = refreshSavedHikesCacheAsync()
+        val success = refreshSavedHikesCacheAsync(forceOverwriteHikesList = false)
```

Comment:
Question: Could you explain why in this case `forceOverwriteHikeList` is set to false?

Reply:
This is related to the `LoadedHikes` enum. This enum is used to maintain state about what type of hikes are currently loaded in the hikes list.

- If we are on the saved hikes screen, that state will be `LoadedHikes.FromSaved`
- If we are in the map screen and we searched for hikes, it will be `LoadedHikes.FromBounds`

Now, even when we are on the map screen, we want to load the saved hikes in the background, to know which ones of the OSM hikes are saved and which ones are not.

To do this, the view model keeps a cache of the user's saved hikes at all times. `refreshSavedHikesCache` is used to load the saved hikes again, but not necessarily overwrite the current list of hikes.

For example, we searched for hikes on the map and we want to see which ones are saved, but we are not sure our local cache of saved hikes is up-to-date. We can call `refreshSavedHikesCache`, and it will refresh the cache without unloading the searched hikes.

Now, sometimes you want to actually overwrite the list of loaded hikes with the saved hikes (when you go on the Saved Hikes screen). In those times you call `loadSavedHikes`, which sets `forceOverwriteHikesList` to `true`.

Here, it's just a call to `refreshSavedHikesCache`, so we don't want to overwrite the existing hikes if any. So we set it to `false`.

Sorry for the lengthy explanation, hope that clears things out!

Reply:
Thanks for adding the documentation for this since it wasn't really clear. Thanks for the explanation the implementation looks good to me.